### PR TITLE
[quiver-team] Add Quiver example for multi-gpu training using GraphSage

### PR DESCRIPTION
**Pull request by eedalong**
[quiver-team] Add Quiver example for multi-gpu training using GraphSage.

### CODE REVIEW

**Comment by luomai** [comment]
Hi PyG team, 

This is the PR to contribute PyG examples with GPU-based sampling + feature aggregation on a GPU and multiple GPUs through Quiver. We observe substantial speed up in the PyG examples with GraphSage. 

We will create another PR for distributed sampling + feature aggregation in 2 - 3 weeks. We have been testing the performance of distributed sampling in the last few weeks, and the code will be ready soon. 

Quiver team.

File: examples/multi_gpu/distributed_sampling_quiver.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+if __name__ == '__main__':
+    dataset = Reddit('../../data/Reddit')
+    data = dataset[0]
+    world_size = torch.cuda.device_count()
+    print('Let\'s use', world_size, 'GPUs!')
+
+    #####################################
+    # Create Quiver Sampler And Feature
+    #####################################
+    csr_topo = quiver.CSRTopo(data.edge_index)
```
Comment (rusty1s) [inline examples/multi_gpu/distributed_sampling_quiver.py:130]: Can we add a link to Quiver documentation here?

**Review by rusty1s — APPROVED** [review]
I think this looks great. Thanks for the efforts. It might be also a good idea to link to this example from the `README.md`. WDYT?

File: examples/multi_gpu/distributed_sampling_quiver.py (hunk 1 above)
Comment (eedalong) [reply examples/multi_gpu/distributed_sampling_quiver.py:130]: OK~

**Comment by luomai** [comment]
> I think this looks great. Thanks for the efforts. It might be also a good idea to link to this example from the `README.md`. WDYT?

@rusty1s Yes, this a good idea! Where shall we put this link? Maybe under the "Implemented GNN models"?

File: examples/multi_gpu/distributed_sampling_quiver.py (hunk 1 above)
Comment (luomai) [reply examples/multi_gpu/distributed_sampling_quiver.py]: The link is now added.

**Comment by eedalong** [comment]
I think this pr is ready to merge, what you guys think? @luomai @rusty1s

**Review by rusty1s — APPROVED** [review]

**Comment by rusty1s** [comment]
Thank you very much! This is great.

**Pull request merged**